### DISCUSSION TITLE
For legacy environments: LDAP authentication and authorization

### DISCUSSION THREAD

**Discussion by unixerius** [Syslifters/sysreptor#267] [General]
Good day!

Looking at some of my customers' local networks, it would be great if Sysreptor could also integrate with classic on-prem Active Directory (and generic Unix environments) via LDAP. 

User accounts would be based on LDAP user objects. Role membership can be tied to specific group membership. 

Of course I don't know how popular such a feature would be, but at least for some folks it might be just what they need.

**Comment by aronmolnar** [comment]
Hi,

Can you explain your specific use case?
Would you like to run SysReptor at your customers' premises?

**Reply by unixerius** [reply]
Correct, yes. And I'm currently running a PoC in my own lab.

I have a number of customers for whom I'm starting/running AppSec programs. As part of this, we are also setting up more formalized internal pentesting procedures. 

It's safe to assume that most customers have Active Directory, LDAP, RedHat IPA or something while not everyone will have an OIDC provider. I know it's becoming rare, but 🤷🏻‍♀️

I'm not saying LDAP is a must-have, but it might make a nice lower barrier of entry for RBAC than having to first make sure you have an OAuth/OIDC setup as well.

For my own lab, I'll look into how easy it's to setup KeyCloak with on-prem Active Directory. But I can't expect that of every customer. 

Of course, if I'm the only one asking for this, the ROI for you will be very low. :)
